### DISCUSSION TITLE
When I install a new npm module, is it normal for expo to quit and have to restar the server?

### DISCUSSION THREAD

**Discussion by 1mike12** [react-native-community/discussions-and-proposals#422] [Ideas]
I'm sure this info is out there but I can't seem to find the right words to google. It seems like every time you either add or remove things in node_modules, the server stops and you have to relaunch the dev process. Normally it's not that big of a deal but I recently had to keep checking out between two different commits with different package.json dependencies which meant I would waste tons of time restarting everything after each checkout. 

If there _was_ some config or something where I didn't have to do this then it would be a lifesaver.

**Comment by gwmccull** [comment]
we use an npm script that deletes `node_modules`, installs the packages and starts the server which makes it a pretty easy "one-step" command

However, I know that I can copy code into my `node_modules` without stopping & starting the server. We have a component library that we use with our app, so when I'm testing changes to the library, I'll copy them over and the bundler just picks them up, although it does do a full refresh of the simulator

You'll want to look in the [Facebook Metro docs](https://facebook.github.io/metro/) for config options. Metro can be set up to use watchman to watch for file changes (I forget if that's still the default setup but it was at one time), so you may need to look at the watchman docs also

**Comment by kelset** [comment]
👋 there - as stated in the README and the issue template, this repo is for long-form discussions. Yours seems more a question for [StackOverflow instead](https://stackoverflow.com/questions/tagged/react-native).
